projects/driver_controls/src/digital_io.c

```diff
@@ -29,100 +30,89 @@ typedef struct DigitalIOData {
 typedef struct DigitalIOSettings {
   GPIOAddress address;
   InterruptEdge edge;
-  DigitalIOData *data;
 } DigitalIOSettings;
 
 // Index the objects using their respective pins
 static DigitalIOData s_input_data[DRIVER_IO_NUM_ADDRESSES];
 
 // Genarate the event based on the identity of the triggering device
-static uint16_t prv_get_event(DigitalIOData *digital_io_data, GPIOState state) {
-  uint16_t event_id = digital_io_data->event;
+static uint16_t prv_get_event(DigitalIOData *data, GPIOState state) {
+  uint16_t event_id = data->event;
 
-  if (event_id == DIGITAL_IO_DEVICE_DIRECTION_SELECTOR && state == GPIO_STATE_HIGH) {
+  if (data->id == DIGITAL_IO_DEVICE_DIRECTION_SELECTOR && state == GPIO_STATE_HIGH) {
     event_id = INPUT_EVENT_DIRECTION_SELECTOR_NEUTRAL;
-  } else if (event_id == DIGITAL_IO_DEVICE_TURN_SIGNAL && state == GPIO_STATE_HIGH) {
+  } else if (data->id == DIGITAL_IO_DEVICE_TURN_SIGNAL && state == GPIO_STATE_LOW) {
```

Comment:
Okay you should probably document why this is low and not high. I assumed these were active-low.

Reply:
That one was a mistake